src/controllers/TalksController.php

```diff
@@ -215,16 +211,11 @@ public function deleteAction($request, $db)
      */
     public function addTrackToTalk(Request $request, PDO $db)
     {
-        if (!isset($request->user_id)) {
-            throw new Exception("You must be logged in to create data", 400);
-        }
+        $this->checkLoggedIn($request);
 
-        $talk_id = $this->getItemId($request);
         $talk_mapper = new TalkMapper($db, $request);
-        $talk = $talk_mapper->getTalkById($talk_id);
-        if (!$talk) {
-            throw new Exception("Talk not found", 404);
```

Comment:
Why do you need to remove this line? Is that relevant to the talk-links?

Removing it has two downsides: 

1. line 218 might fail when the user asks for a talk not available
2. The user will never know that the talk doesn't exist.